clang/include/clang/AST/ASTContext.h

```diff
@@ -182,6 +182,17 @@ struct TypeInfoChars {
   }
 };
 
+// Interface that allows constant evaluator to mutate AST.
+// Sema is the only entity that can implement this.
+struct EvalASTMutator {
+  virtual ~EvalASTMutator() = default;
+
+  virtual void
+  InstantiateFunctionDefinition(SourceLocation PointOfInstantiation,
+                                FunctionDecl *Function, bool Recursive,
+                                bool DefinitionRequired, bool AtEndOfTU) = 0;
```

Comment:
```suggestion
                                FunctionDecl *Function) = 0;
```

I don't think this will ever be useful

Reply:
Fixed